### PR TITLE
[5.2] Fixing typehint for getListQuery to QueryInterface

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
The database object in Joomla returns an object implementing QueryInterface, but all our docblocks for getListQuery claim it is a DatabaseQuery object. This PR unifies this to use the QueryInterface everywhere. This set of changes is the result of a run of phpstan on our codebase.

### Testing Instructions
Codereview.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: components/com_banners/src/Model/BannersModel.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ protected function getStoreId($id = '')
     /**
      * Method to get a DatabaseQuery object for retrieving the data set from a database.
      *
-     * @return  DatabaseQuery   A DatabaseQuery object to retrieve the data set.
+     * @return  QueryInterface   A DatabaseQuery object to retrieve the data set.
```
Comment (Quy) [inline components/com_banners/src/Model/BannersModel.php]: Update description?

File: administrator/components/com_users/src/Model/NotesModel.php [hunk 2]
```diff
@@ -62,7 +62,7 @@ public function __construct($config = [], MVCFactoryInterface $factory = null)
     /**
      * Build an SQL query to load the list data.
      *
-     * @return  DatabaseQuery  A DatabaseQuery object to retrieve the data set.
+     * @return  QueryInterface  A DatabaseQuery object to retrieve the data set.
```
Comment (Quy) [inline administrator/components/com_users/src/Model/NotesModel.php]: Update description?

File: administrator/components/com_privacy/src/Model/RequestsModel.php [hunk 3]
```diff
@@ -52,7 +52,7 @@ public function __construct($config = [])
     /**
      * Method to get a DatabaseQuery object for retrieving the data set from a database.
```
Comment (Quy) [inline administrator/components/com_privacy/src/Model/RequestsModel.php]: Update?

File: administrator/components/com_privacy/src/Model/ConsentsModel.php [hunk 4]
```diff
@@ -55,7 +55,7 @@ public function __construct($config = [])
     /**
      * Method to get a DatabaseQuery object for retrieving the data set from a database.
```
Comment (Quy) [inline administrator/components/com_privacy/src/Model/ConsentsModel.php]: Update?

File: administrator/components/com_fields/src/Model/GroupsModel.php [hunk 5]
```diff
@@ -119,7 +120,7 @@ protected function getStoreId($id = '')
     /**
      * Method to get a DatabaseQuery object for retrieving the data set from a database.
```
Comment (Quy) [inline administrator/components/com_fields/src/Model/GroupsModel.php]: Update?

File: administrator/components/com_fields/src/Model/FieldsModel.php [hunk 6]
```diff
@@ -131,7 +132,7 @@ protected function getStoreId($id = '')
     /**
      * Method to get a DatabaseQuery object for retrieving the data set from a database.
```
Comment (Quy) [inline administrator/components/com_fields/src/Model/FieldsModel.php]: Update?

File: components/com_banners/src/Model/BannersModel.php (hunk 1 above)
Comment (Hackwar) [reply components/com_banners/src/Model/BannersModel.php]: done

File: administrator/components/com_privacy/src/Model/RequestsModel.php (hunk 3 above)
Comment (Hackwar) [reply administrator/components/com_privacy/src/Model/RequestsModel.php]: Done.

File: administrator/components/com_users/src/Model/NotesModel.php (hunk 2 above)
Comment (Hackwar) [reply administrator/components/com_users/src/Model/NotesModel.php]: Done.

File: administrator/components/com_privacy/src/Model/ConsentsModel.php (hunk 4 above)
Comment (Hackwar) [reply administrator/components/com_privacy/src/Model/ConsentsModel.php]: Done.

File: administrator/components/com_fields/src/Model/GroupsModel.php (hunk 5 above)
Comment (Hackwar) [reply administrator/components/com_fields/src/Model/GroupsModel.php]: Done.

File: administrator/components/com_fields/src/Model/FieldsModel.php (hunk 6 above)
Comment (Hackwar) [reply administrator/components/com_fields/src/Model/FieldsModel.php]: Done.

**Review by rdeutz — APPROVED** [review]

File: administrator/components/com_fields/src/Model/FieldsModel.php (hunk 6 above)
Comment (pe7er) [inline administrator/components/com_fields/src/Model/FieldsModel.php]: @Hackwar Could you please change this to:

* Method to get a QueryInterface object for retrieving the data set from a database.

File: administrator/components/com_fields/src/Model/GroupsModel.php [hunk 7]
```diff
@@ -117,9 +118,9 @@ protected function getStoreId($id = '')
     }
 
     /**
-     * Method to get a DatabaseQuery object for retrieving the data set from a database.
+     * Method to get an object implementing QueryInterface for retrieving the data set from a database.
```
Comment (pe7er) [inline administrator/components/com_fields/src/Model/GroupsModel.php]: @Hackwar please change to:

* Method to get a QueryInterface object for retrieving the data set from a database.

File: components/com_banners/src/Model/BannersModel.php [hunk 8]
```diff
@@ -58,7 +58,7 @@ protected function getStoreId($id = '')
     /**
      * Method to get a DatabaseQuery object for retrieving the data set from a database.
```
Comment (pe7er) [inline components/com_banners/src/Model/BannersModel.php]: @Hackwar please change to:

* Method to get a QueryInterface object for retrieving the data set from a database.

File: administrator/components/com_fields/src/Model/GroupsModel.php [hunk 9]
```diff
@@ -117,9 +118,9 @@ protected function getStoreId($id = '')
...
      *
-     * @return  \Joomla\Database\DatabaseQuery   A DatabaseQuery object to retrieve the data set.
+     * @return  QueryInterface   A DatabaseQuery object to retrieve the data set.
```
Comment (pe7er) [inline administrator/components/com_fields/src/Model/GroupsModel.php]: @Hackwar please change to:

* @return QueryInterface An object implementing QueryInterface to retrieve the data set.

**Pull request merged**